core/app/[locale]/(default)/account/wishlists/[id]/page.tsx

```diff
@@ -0,0 +1,107 @@
+import { notFound } from 'next/navigation';
+import { getFormatter, getTranslations } from 'next-intl/server';
+import { SearchParams } from 'nuqs';
+import { createSearchParamsCache, parseAsInteger, parseAsString } from 'nuqs/server';
+
+import { CursorPaginationInfo } from '@/vibes/soul/primitives/cursor-pagination';
+import { Wishlist, WishlistDetails } from '@/vibes/soul/sections/wishlist-details';
+import { ExistingResultType } from '~/client/util';
+import { defaultPageInfo, pageInfoTransformer } from '~/data-transformers/page-info-transformer';
+import { wishlistDetailsTransformer } from '~/data-transformers/wishlists-transformer';
+import { isMobileUser } from '~/lib/user-agent';
+
+import { removeWishlistItem } from '../_actions/remove-wishlist-item';
+import { getDeleteWishlistModal, getRenameWishlistModal } from '../modals';
+
+import { addWishlistItemToCart } from './_actions/add-to-cart';
+import { WishlistActions, WishlistActionsSkeleton } from './_components/wishlist-actions';
+import { getCustomerWishlist } from './page-data';
+
+interface Props {
+  params: Promise<{ id: string }>;
+  searchParams: Promise<SearchParams>;
+}
+
+const defaultWishlistItemsLimit = 10;
+const searchParamsCache = createSearchParamsCache({
+  tag: parseAsString,
+  before: parseAsString,
+  after: parseAsString,
+  limit: parseAsInteger.withDefault(defaultWishlistItemsLimit),
+});
+
+async function getWishlist(
+  id: string,
+  t: ExistingResultType<typeof getTranslations>,
+  pt: ExistingResultType<typeof getTranslations>,
+  searchParamsPromise: Promise<SearchParams>,
+): Promise<Wishlist> {
+  const entityId = Number(id);
+  const searchParamsParsed = searchParamsCache.parse(await searchParamsPromise);
+  const formatter = await getFormatter();
+  const wishlist = await getCustomerWishlist({ entityId, ...searchParamsParsed });
+
+  if (!wishlist) {
+    return notFound();
+  }
+
+  return wishlistDetailsTransformer(wishlist, t, pt, formatter);
```

Comment:
I'm not sure how big of a fan I am of passing the translations getters to the transformer. Is there a big performance benefit over just calling these in the transformer directly?

Reply:
@jorgemoya This is a similar pattern we follow for the Orders transformer. Since we've already fetched the translations in the page, I figured it made sense to just pass them down instead of turning the transformer into an `async` function.

Since transformers' sole purpose is to just convert model A into model B, I personally feel like they shouldn't be async at all 💭  We can discuss this more though!